internal/providers/terraform/google/sql_database_instance.go

```diff
@@ -1,305 +1,38 @@
-	dbTypeString := sqlInstanceTypeToDescriptionName(dbType)
-	availabilityTypeString := availabilityTypeDescName(availabilityType)
-
-	description := fmt.Sprintf("%s: %s", dbTypeString, availabilityTypeString)
-
-	return description
-}
-
-func sqlInstanceStorage(region string, dbType SQLInstanceDBType, availabilityType, diskType string, diskSizeGB int64) *schema.CostComponent {
-	diskTypeHumanReadableNames := map[string]string{
-		"PD_SSD": "SSD",
-		"PD_HDD": "HDD",
+	if !d.IsEmpty("replica_configuration") {
+		r.ReplicaConfiguration = d.Get("replica_configuration").String()
 	}
-
-	diskTypeAPIResourceGroup := map[string]string{
-		"PD_SSD": "SSD",
-		"PD_HDD": "PDStandard",
+	if !d.IsEmpty("settings.0.availability_type") {
+		r.AvailabilityType = d.Get("settings.0.availability_type").String()
 	}
-
-	if dbType == SQLServer {
-		diskType = "PD_SSD"
+	if !d.IsEmpty("settings.0.disk_type") {
+		r.DiskType = d.Get("settings.0.disk_type").String()
 	}
-
-	cost := &schema.CostComponent{
-		Name:            fmt.Sprintf("Storage (%s, %s)", diskTypeHumanReadableNames[diskType], strings.ToLower(availabilityType)),
-		Unit:            "GB",
-		UnitMultiplier:  decimal.NewFromInt(1),
-		MonthlyQuantity: decimalPtr(decimal.NewFromInt(diskSizeGB)),
-		ProductFilter: &schema.ProductFilter{
-			VendorName: strPtr("gcp"),
-			Region:     strPtr(region),
-			Service:    strPtr("Cloud SQL"),
-			AttributeFilters: []*schema.AttributeFilter{
-				{Key: "resourceGroup", Value: strPtr(diskTypeAPIResourceGroup[diskType])},
-				{Key: "description", ValueRegex: strPtr(fmt.Sprintf("/%s: %s/", sqlInstanceTypeToDescriptionName(dbType), availabilityTypeDescName(availabilityType)))},
-			},
-		},
-	}
-
-	return cost
-}
-
-func backupCostComponent(region string, quantity *decimal.Decimal) *schema.CostComponent {
-	cost := &schema.CostComponent{
-		Name:            "Backups",
-		Unit:            "GB",
-		UnitMultiplier:  decimal.NewFromInt(1),
-		MonthlyQuantity: quantity,
-		ProductFilter: &schema.ProductFilter{
-			VendorName: strPtr("gcp"),
-			Region:     strPtr(region),
-			Service:    strPtr("Cloud SQL"),
-			AttributeFilters: []*schema.AttributeFilter{
-				{Key: "resourceGroup", Value: strPtr("PDSnapshot")},
-				{Key: "description", ValueRegex: strPtr("/Cloud SQL: Backups/")},
-			},
-		},
-	}
-
-	return cost
-}
-
-func ipv4CostComponent() *schema.CostComponent {
-	return &schema.CostComponent{
-		Name:           "IP address (if unused)",
-		Unit:           "hours",
-		UnitMultiplier: decimal.NewFromInt(1),
-		HourlyQuantity: decimalPtr(decimal.NewFromInt(1)),
-		ProductFilter: &schema.ProductFilter{
-			VendorName: strPtr("gcp"),
-			Region:     strPtr("global"),
-			Service:    strPtr("Cloud SQL"),
-			AttributeFilters: []*schema.AttributeFilter{
-				{Key: "resourceGroup", Value: strPtr("IpAddress")},
-				{Key: "description", ValueRegex: strPtr("/IP address idling - hour/")},
-			},
-		},
+	if !d.IsEmpty("settings.0.disk_size") {
```

Comment:
As above, I don't think this empty check is needed, since `.Int()` will default to 0, which is what `DiskSize` defaults to anyway.

Reply:
Applied